ios/RNSScreen.mm

```diff
@@ -1031,6 +1038,20 @@ - (void)viewDidLayoutSubviews
   }
 }
 
+- (BOOL)assertModalHierarchy
+{
+  if (self.childViewControllers.count == 0) {
+    RCTLogError(
+        @"Modal has been rendered without the navigation controller, which is not allowed. Ensure that modal has RNSNavigationController mounted as a child view controller.");
+  } else if (self.childViewControllers.count > 1) {
+    RCTLogError(
+        @"Modal has been rendered with more than one navigation controller, which is not allowed. Ensure that modal has only one RNSNavigationController mounted as a child view controller.");
+  }
+
+  return self.childViewControllers.count == 1 &&
+      [self.childViewControllers[0] isKindOfClass:UINavigationController.class];
+}
+
```

Comment:
Hey, as I've shown in last review & as we discussed it is possible that "modal" view controller hierarchy is comprised of only single `RNSScreen` (in case it has no header).

Thus these log messages might be misleading & incorrect.

Reply:
Fixed in https://github.com/software-mansion/react-native-screens/pull/1917/commits/e779764a5423b326af1d71e2cbc3da4cfd5ca201.